### PR TITLE
Fixes #312 make implicit uriParameters, explicit.

### LINKED ISSUE #312: URI parameters are not extracted if not formally defined
Hello,

**Parser: raml-java-parser2**

Use case:

```
  /users:
    get:
      description: get users
      responses:
        200:
          body:
            application/json:
  /users/{user_id}:
    description:
    get:
      description: one user
      responses:
        200:
```

With example.raml, the following is parsed with a /users/{user_id} resource that has with an empty uriParameters():

According to the spec:

> If a URI parameter in a resource's relative URI is not explicitly described in a uriParameters property for that resource, it MUST still be treated as a URI parameter with defaults as specified in the Named Parameters section of this specification. Its type is "string", it is required, and its displayName is its name (i.e. without the surrounding curly brackets [{] and [}]).

So I'm expecting that uriParameters() will return the inferred URI parameter definition using the specified default values.

### PR DESCRIPTION
**Pull request by pacafuentes**

### CODE REVIEW

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    public boolean matches(Node node)
+    {
+        return node instanceof ResourceNode;
+    }
+
+    @Override
+    public Node transform(Node node)
+    {
+        List<String> templates = getUriTemplates(((ResourceNode) node).getRelativeUri());
+        injectImplicitUriParameters(((ResourceNode) node).getValue(), templates);
```
Comment (dfmacias) [inline raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java]: you are casting node twice

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @Override
+    public boolean matches(Node node)
+    {
+        return node instanceof ResourceNode;
+    }
+
+    @Override
+    public Node transform(Node node)
+    {
+        List<String> templates = getUriTemplates(((ResourceNode) node).getRelativeUri());
```
Comment (dfmacias) [inline raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java]: `getUriTemplates` is a method of another transformer. Maybe a better approach would be moving it to a util class. What do you think?

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        else if (!templates.isEmpty())
+        {
+            StringNodeImpl uriParameters = new StringNodeImpl(URI_PARAMETERS);
+            ObjectNodeImpl objectNode = new ObjectNodeImpl();
+
+            addUriParameters(templates, objectNode);
+            node.addChild(new KeyValueNodeImpl(uriParameters, objectNode));
+        }
+
+        if (parametersNode != null && !templates.isEmpty())
```
Comment (dfmacias) [inline raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java]: This block could be inside of the first condition

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java (hunk 1 above)
Comment (pacafuentes) [reply raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java]: done

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java (hunk 2 above)
Comment (pacafuentes) [reply raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java]: done

File: raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java (hunk 3 above)
Comment (pacafuentes) [reply raml-parser-2/src/main/java/org/raml/v2/internal/impl/v10/phase/ImplicitUriParametersInjectionTransformer.java]: done

**Pull request merged**